docs/developer/logging.md

```diff
@@ -0,0 +1,40 @@
+# Logging
+
+This document describes how the EDC code should be logged and why.
+
+## Logging component
+
+Logs can be produced using the [`Monitor`](../../spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/monitor/Monitor.java) service, 
+that offers 4 different log levels:
+
+### `severe` 
+> Error events that might lead the application to abort or still allow it to continue running.
+
+Used in case of an unexpected interruption of the flow.
+
+### `warning`
+> Potentially harmful situations messages.
+
+Used in case of an expected event that does not interrupt the flow but that should be taken into consideration. 
+ 
+### `info`
+> Informational messages that highlight the progress of the application at coarse-grained level.
+ 
+Used to describe the normal flow of the application.
+ 
+### `debug` 
+> Fine-grained informational events that are most useful to debug an application.
+ 
+Used to describe details of the normal flow that are not interesting for a production environment.
+
+## What should be logged
+
+- every incoming call with `debug`
+- every state change with `info`
```

Comment:
Shouldn't state changes be debug? Otherwise, these messages will likely drown out other important (non-warning) information.

Reply:
@jimmarino I can agree with this, but what have to be logged with `info` level?

Reply:
For info, I think any important message that is not an error. For example, an extension is installed and booted. Or, a maintenance process is run. 

IMO granular runtime operations in production are better handled by observability/telemetry.  For development, I generally never use logs, just a debugger.

Reply:
> IMO granular runtime operations in production are better handled by observability/telemetry. For development, I generally never use logs, just a debugger

@jimmarino I agree with you that for us developers the debugger is the better tooling. But in our recent rollout of the connector to the first companies of catena-x, we identified that most of the people trying to setup the connector don't have deep insights in devops approaches like observability/telemetry/tracability and therefor weren't able to debug into their connectors which were deployed with k8s or docker-compose (mostly they just opened the port and we edc developer then debugged into the connector). They just looked at the logs and were confused when they don't see an error message (e.g. when they got a 400 for their request).

So I totally agree with @MoritzKeppler that we need to **log as much useful informations as possible**. Therefor maybe we can introduce also a new log level like `audit`, which then is just for state changes and creations, updates, deletion of domain entities (e.g. via management api).

Reply:
I don't think state changes should be logged as info. That will pollute the logs with useless information in production deployments.

+1 for auditing, but that should be done via architected events. @ndr-brt is working on an eventing subsystem that can be used for that.

@florianrusch-zf The issue you identify appears to be twofold:

1. Developers should know how to use a debugger. Relying on logs as a substitute isn't a maintainable or effective solution.
2. The API should send back reasonable error information, preferrably formatted as JSON. It looks like that did not happen and would be worth filing a separate issue.

Reply:
There are great tools to filter and aggregate logs, so I wouldn't fear the amount of logs lines. 
Analysis later in time cannot be done by using a debugger. And the connector will not only be run by developers.
Log pollution will be caused by inconsistent or chatty loglines, this is something we need to avoid.

Reply:
@MoritzKeppler introducing a filtering tool IMO just papers over the problem.  I don't think it makes sense to log debug information under "INFO". If we log in under DEBUG, people can always choose to enable that level and use filtering tools to sift through logs.